### PR TITLE
Restrict target configs keys

### PR DESCRIPTION
**Pull request by ahuang11**
Decouples external collections blocks' fields from dbt configs (e.g. if a new field is added, won't inadvertently also add it to the dbt profile unless it's an expected key).

Also, user was experiencing issue with async so dropped the async. I don't think async matters much here anyways.

Closes https://github.com/PrefectHQ/prefect-dbt/issues/104 and https://github.com/PrefectHQ/prefect-dbt/issues/108

### Checklist

- [x] References any related issue by including "Closes #<Issue Number>" or "Closes <Issue URL>".
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect-dbt/issues/new/choose) first.
- [x] Includes tests or only affects documentation.
- [x] Passes `pre-commit` checks.
  - Run `pre-commit install && pre-commit run --all` locally for formatting and linting.
- [x] Includes screenshots of documentation updates.
  - Run `mkdocs serve` view documentation locally.
- [x] Summarizes PR's changes in [CHANGELOG.md](https://github.com/PrefectHQ/prefect-dbt/blob/main/CHANGELOG.md)

### CODE REVIEW

**Comment by ahuang11** [comment]
Not doing Postgres here because that requires a much larger effort to pivot towards the new SqlAlchemyConnector, and potentially breaking (or lots of deprecation to handle)

**Comment by ahuang11** [comment]
Tests will fail until https://github.com/PrefectHQ/prefect-snowflake/pull/58 is merged.

File: prefect_dbt/cli/configs/bigquery.py [hunk 1]
```diff
@@ -86,8 +87,7 @@ class BigQueryTargetConfigs(TargetConfigs):
     project: Optional[str] = None
     credentials: GcpCredentials
 
-    @sync_compatible
-    async def get_configs(self) -> Dict[str, Any]:
+    def get_configs(self) -> Dict[str, Any]:
```
Comment (desertaxle) [inline prefect_dbt/cli/configs/bigquery.py:90]: Is this technically a breaking change? Would you see errors if you were previously awaiting this method?

Comment (ahuang11) [reply prefect_dbt/cli/configs/bigquery.py:90]: Yes this is a breaking change

**Review by desertaxle — APPROVED** [review]

**Pull request merged**